Add __repr__ methods to DIDL classes for easier debugging

chishm opened the pull request:
The `__repr__` methods return a string that can be `eval()`ed to recreate the object (this is checked with tests), showing how the object could be constructed.

I've also dropped the `didl_lite:res` property def from the base `DidlObject` because it was completely subsumed by the explicit `DidlObject.resources` property, so didn't make any sense in the repr output.

StevenLooman commented:
Thank you for your PR! Looks good to me except for the dropped `didl_lite:res` property. Although it might not be used, it came from a document describing all the properties of the items, but I cannot find the document itself. I'd rather stick to the 'formal' specification.

I'll do some more searching hoping I can find it again.

StevenLooman commented:
See [UPnP-av-ContentDirectory-v2-Service](http://upnp.org/specs/av/UPnP-av-ContentDirectory-v2-Service.pdf]), page 116, section B.1.9.

So perhaps the `DidlObject.resources` property should be removed and `DidlObject.res` should be used instead.

StevenLooman commented:
Fixed the conflict.

chishm commented:
Thanks for having a good look at this PR. I've redone the `didl_lite:res` stuff so it now points at the `DidlObject.resources` attribute.
This means the API is not broken for any users of the class.

If you're happy with this approach, I'll rebase this branch and squash the fixup commits.

chishm commented:
I see you've also fixed this up on master, at almost the same time I updated this PR :rofl: 

I can rebase and remove the didl_lite:res commits, if you prefer.

StevenLooman commented:
Hehe, sorry about that. :) Lets see.

StevenLooman commented:
> I can rebase and remove the didl_lite:res commits, if you prefer.

Please do this. I'd rather stick to the regular way of attributes than to make an exception for resources.

chishm commented:
Ok, this should be good to go.

StevenLooman commented:
Can you add a line to the `CHANGES.rst` file?

StevenLooman commented:
Now you've triggered me about `desc`/descriptors.

StevenLooman commented:
Now you've triggered me about `desc`/descriptors. I'll take a look at it later.

StevenLooman commented:
Thank you for this PR! Do you want a release for this?

chishm commented:
Thank you for the merge! No release is necessary, I'm just sharing what I'm using during development.

> Now you've triggered me about `desc`/descriptors. I'll take a look at it later.

That's why it took me a little longer to do the `res`/`resources` alias stuff. I was trying to figure out how to match it to `desc`/`descriptors`, but couldn't find anything useful in the ContentDirectory spec about `desc`, and the XSD doesn't follow the DIDL object model.

StevenLooman commented:
I couldn't find anything either, probably the reason it is an explicit parameter. Thank you for looking into it and describing your results. I came to the same conclusion. Lets keep `desc` for now.
